components/FlexibleHeader/src/MDCFlexibleHeaderView.m

```diff
@@ -1431,30 +1457,58 @@ - (void)setTrackingScrollView:(UIScrollView *)trackingScrollView {
     [self fhv_startObservingContentOffset];
   }
 
-  // When canAlwaysExpandToMaximumHeight is enabled our header's height no longer directly
-  // correlates to the content offset - it's also augmented by the shift accumulator. In order to
-  // keep the header's height constant when changing the tracking scroll view, we need to adjust
-  // the shift accumulator accordingly.
-  if (self.canAlwaysExpandToMaximumHeight && self.sharedWithManyScrollViews &&
-      wasTrackingScrollView) {
+  BOOL shouldAnimate = NO;
+
+  if (self.sharedWithManyScrollViews && wasTrackingScrollView) {
     // What's our expected height now that we've changed the tracking scroll view?
     CGFloat headerHeight = -[self fhv_contentOffsetWithoutInjectedTopInset];
     headerHeight = MAX(self.computedMinimumHeight, MIN(self.computedMaximumHeight, headerHeight));
 
     // How much will our height change if we do nothing right now?
-    const CGFloat heightDelta = self.bounds.size.height - headerHeight;
+    const CGFloat heightDelta = stashedHeight - headerHeight;
+
+    // When canAlwaysExpandToMaximumHeight is enabled our header's height no longer directly
+    // correlates to the content offset - it's also augmented by the shift accumulator. In order to
+    // keep the header's height constant when changing the tracking scroll view, we need to adjust
+    // the shift accumulator accordingly.
+    if (self.canAlwaysExpandToMaximumHeight) {
+      // Cap the accumulator to ensure it's valid.
+      CGFloat accumulatorMin;
+      if (headerHeight > self.computedMinimumHeight + DBL_EPSILON) {
```

Comment:
Why use DBL_EPSILON? The value might actually be unrepresentable on 32-bit hosts. Can we use something like `kContentOffsetEpsilon` instead?